modules/javafx.graphics/src/main/java/javafx/application/PlatformPreferences.java

```diff
@@ -0,0 +1,202 @@
+ *         <tr><td>{@code Windows.SysColor.COLOR_HIGHLIGHTTEXT}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_HOTLIGHT}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_INACTIVEBORDER}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_INACTIVECAPTION}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_INACTIVECAPTIONTEXT}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_INFOBK}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_INFOTEXT}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_MENU}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_MENUHILIGHT}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_MENUBAR}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_MENUTEXT}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_SCROLLBAR}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_WINDOW}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_WINDOWFRAME}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.SysColor.COLOR_WINDOWTEXT}</td><td>{@link Color}</td></tr>
+ *     </table>
+ *     <p>
+ *     <table border="1" style="table-layout: fixed; width: 100%; max-width: 40em">
+ *         <caption style="text-align: left">Theme colors, reported by {@code Windows.UI.ViewManagement.UISettings}</caption>
+ *         <tr><td>{@code Windows.UIColor.Background}</td><td style="width: 5em">{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.UIColor.Foreground}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.UIColor.AccentDark3}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.UIColor.AccentDark2}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.UIColor.AccentDark1}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.UIColor.Accent}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.UIColor.AccentLight1}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.UIColor.AccentLight2}</td><td>{@link Color}</td></tr>
+ *         <tr><td>{@code Windows.UIColor.AccentLight3}</td><td>{@link Color}</td></tr>
+ *     </table>
+ *     Note that {@code Windows.UI.ViewManagement.UISettings} is available since Windows 10 build 10240.
+ *     <p>
+ *
+ *     <li><span style="font-weight: bold">macOS</span>
+ *         <p>Platform preferences are currently not reported on macOS.
+ *
+ *     <li><span style="font-weight: bold">Linux</span>
+ *         <p>Platform preferences are currently not reported on Linux.
+ * </ol>
+ *
+ * @since 20
+ */
+public interface PlatformPreferences extends Map<String, Object> {
```

Comment:
Are you sure it is a good idea to expose these as a `Map`?  It seems to me that this isn't that good a practice any more to have classes implement or extend lists/maps/sets as they pull in a huge amount of API surface that is mostly useless or too general for the class's purpose.

The addition of the listener management methods also has me wondering, as `ObservableMap` does something similar.

Reply:
All of the mutating methods are useless, since the implementation always returns a read-only map. However, the alternative would be to duplicate APIs to enumerate and inspect the contents of the `PlatformPreferences` map (applications might want to show a list of available preferences). I'm not sure that's preferable, mostly because `PlatformPreferences` _does_ represent a mapping of keys to values.

It's true that listener management makes it look like an `ObservableMap`. The difference is that `ObservableMap` doesn't support batch change notifications, which the current implementation relies on to minimize the number of style theme resets. Of course, that could be solved in a different way.

Reply:
> The addition of the listener management methods also has me wondering, as `ObservableMap` does something similar.

I've switched to using `ObservableMap`.